sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +374,201 @@ def transform(self, X):
+        """
+        self._validate_params()
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64,
+                        force_all_finite=False)
+        self.n_features_ = X.shape[1]
+
+        if isinstance(self.features, six.string_types):
+            if self.features == "auto":
+                _, self.feat_with_missing_ = self._get_missing_features_info(X)
+            else:  # self.features == "all"
+                self.feat_with_missing_ = np.arange(self.n_features_)
+        else:
+            self.feat_with_missing_ = self.features
+
+        return self
+
+    def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        check_is_fitted(self, "feat_with_missing_", "n_features_")
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64,
+                        force_all_finite=False)
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different number of features "
+                             "than during fitting.")
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        if isinstance(self.features, six.string_types):
+            if self.features == "auto":
+                features = np.setdiff1d(feat_with_missing,
+                                        self.feat_with_missing_)
+                if self.error_on_new and features.size > 0:
+                    raise Exception("The features %s have missing values "
+                                    "in transform but have no missing values "
+                                    "in fit" % features)
+
+        if not (isinstance(self.features, six.string_types) and
+                self.features == "all") and len(self.feat_with_missing_) != 0:
+            # no need to slice when all features have missing values
+            imputer_mask = imputer_mask[:, self.feat_with_missing_]
+
+        return imputer_mask
+
+    def fit_transform(self, X, y=None):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        self.fit(X, y)
+        return self.transform(X)
+
+    def _validate_params(self):
+        if (isinstance(self.features, six.string_types) and
+                self.features not in ["auto", "all"]):
+            raise ValueError("Can only use these options: 'auto', 'all'"
```

Comment:
These validation errors are not tested.

Reply:
Have added tests for these validations now. Thanks.